Sail/src/Sail/entities/systems/network/HostSendToSpectatorSystem.cpp

```diff
@@ -0,0 +1,127 @@
+#include "pch.h"
+
+#include "HostSendToSpectatorSystem.h"
+
+#include "Sail/entities/components/NetworkSenderComponent.h"
+#include "Sail/entities/components/OnlineOwnerComponent.h"
+#include "Sail/entities/components/LocalOwnerComponent.h"
+#include "Sail/entities/Entity.h"
+
+#include "Network/NWrapperSingleton.h"
+#include "Sail/../../libraries/cereal/archives/portable_binary.hpp"
+
+#include "../src/Network/NWrapperSingleton.h"
+
+HostSendToSpectatorSystem::HostSendToSpectatorSystem() {
+	registerComponent<NetworkSenderComponent>(true, true, true);
+	registerComponent<TransformComponent>(true, true, false);
+}
+
+HostSendToSpectatorSystem::~HostSendToSpectatorSystem() {
+}
+
+void HostSendToSpectatorSystem::init(Netcode::PlayerID playerID) {
+	m_playerID = playerID;
+}
+
+
+// The messages this function creates must match the format of messages created by and received by NetworkSenderSystem and NetworkReceiverSystem
+void HostSendToSpectatorSystem::sendEntityCreationPackage(TCP_CONNECTION_ID TCPidPlayer) const {
+	std::ostringstream dataString(std::ios::binary);
+	Netcode::OutArchive ar(dataString);
+
+	// -+-+-+-+-+-+-+-+ Per-frame sends to per-frame receives via components -+-+-+-+-+-+-+-+ 
+	// Send our playerID so that we can ignore this packet when it gets back to us from the host
+	ar(m_playerID);
+
+	// Don't send any data from NetworkSenderComponents.
+	ar(size_t{ 0 });
+
+
+
+	// TODO: if a player's candle isn't held also send an event for that
+	size_t nrOfEvents = 0;
+	bool candleHeld = true;
+
+	// Find how many players are alive so that the receiver knows how many messages to receive
+	for (auto e : entities) {
+		if (e->getComponent<NetworkSenderComponent>()->m_entityType == Netcode::EntityType::PLAYER_ENTITY) {
+			nrOfEvents++;
+
+			for (auto c : e->getChildEntities()) {
+				if (c->hasComponent<CandleComponent>() && !c->getComponent<CandleComponent>()->isCarried) {
```

Comment:
CandleComponent is not a registered component in the constructor.

registerComponent\<CandleComponent\>(false, true, false);